tools/pythonpkg/src/python_replacement_scan.cpp

```diff
@@ -17,6 +17,12 @@ namespace duckdb {
 static void CreateArrowScan(const string &name, py::object entry, TableFunctionRef &table_function,
                             vector<unique_ptr<ParsedExpression>> &children, ClientProperties &client_properties,
                             PyArrowObjectType type) {
+
+	if (type == PyArrowObjectType::PyCapsuleInterface) {
+		entry = entry.attr("__arrow_c_stream__")();
```

Comment:
Small nitpick: no producer should implement it wrongly of course, but it might still be good practice to verify that the returned capsule has the correct name, like you do in `DuckDBPyConnection::GetArrowType` when verifying a raw capsule as input.

Reply:
I believe that's handled at a later point, but I'm not entirely sure

Reply:
That's probably handled here? https://github.com/duckdb/duckdb/pull/13386/files#diff-7c2052a992507edd04315acf93ff1c68628dd402ddd4c0658db600d9af5674aaR1978-R1980